ivy_models/mlpmixer/mlpmixer.py

```diff
@@ -0,0 +1,181 @@
+    def __init__(
+        self,
+        vol=72,
+        inp_dim=3,
+        num_classes=10,
+        num_blocks=2,
+        patch_size=16,
+        hidden_dim=128,
+        token_mlp_dim=64,
+        channel_mlp_dim=32,
+        data_format="NCHW",
+        spec=None,
+        v=None,
+    ):
+        self.spec = (
+            spec
+            if spec and isinstance(spec, MLPMixerSpec)
+            else MLPMixerSpec(
+                vol=vol,
+                inp_dim=inp_dim,
+                num_classes=num_classes,
+                num_blocks=num_blocks,
+                patch_size=patch_size,
+                hidden_dim=hidden_dim,
+                token_mlp_dim=token_mlp_dim,
+                channel_mlp_dim=channel_mlp_dim,
+                data_format=data_format,
+            )
+        )
+        self.num_blocks = num_blocks
+        self.inp_dim = inp_dim
+        self.hidden_dim = hidden_dim
+        self.patch_size = patch_size
+        self.vol = vol
+        self.token_mlp_dim = token_mlp_dim
+        self.channel_mlp_dim = channel_mlp_dim
+        self.num_classes = num_classes
+
+        super(MLPMixer, self).__init__(v=v)
+
+    def _build(self, *args, **kwargs):
+        self.conv = ivy.Conv2D(
+            self.inp_dim,
+            self.hidden_dim,
+            [self.patch_size, self.patch_size],
+            [self.patch_size, self.patch_size],
+            0,
+            data_format="NCHW",
+        )
+        self.num_patches = int((self.vol / self.patch_size) ** 2)
+        self.mixer_block = MixerBlock(
+            self.hidden_dim, self.num_patches, self.token_mlp_dim, self.channel_mlp_dim
+        )
+        self.norm = ivy.LayerNorm([self.hidden_dim])
+        self.dropout = ivy.Dropout(0.25)
+        self.linear = ivy.Linear(self.hidden_dim, self.num_classes)
+
+    @classmethod
+    def get_spec_class(self):
+        return MLPMixerSpec
+
+    def _forward(self, x):
+        x = self.conv(x)
+        x = x.reshape(
+            (int(x.shape[0]), int(x.shape[1]) * int(x.shape[2]), int(x.shape[3]))
+        )
+        for i in range(self.num_blocks):
+            x = self.mixer_block(x)
+        x = self.norm(x)
+        # x = self.dropout(x)
+        x = ivy.mean(x, axis=1)
+        logits = self.linear(x)
+        probs = ivy.softmax(logits)
+
+        return probs
+
+
+def mlpmixer(
+    pretrained=True,
+    vol=72,
+    inp_dim=3,
+    num_classes=10,
+    num_blocks=4,
+    patch_size=9,
+    hidden_dim=128,
+    token_mlp_dim=64,
+    channel_mlp_dim=128,
+):
+    """Ivy MLPMixer model"""
+    model = MLPMixer(
+        vol,
+        inp_dim,
+        num_classes=num_classes,
+        num_blocks=num_blocks,
+        patch_size=patch_size,
+        hidden_dim=hidden_dim,
+        token_mlp_dim=token_mlp_dim,
+        channel_mlp_dim=channel_mlp_dim,
+    )
+    if pretrained:
+        return MLPMixer.load_from_huggingface(
+            repo_id="unifyai/MLPMixer", weights_path="weights.hdf5"
```

Comment:
I've briefly gone through the hf repo, and I think the `config.json` is missing, the required files will be uploaded if the method `push_to_huggingface` is used when uploading the trained model, could you please reupload with that? thanks a lot!